### PR TITLE
[14.0][FIX] hr_expense_advance_clearing: _onchange_advance_sheet_id for hooks

### PR DESCRIPTION
**Pull request by Saran440**
change function `_onchange_advance_sheet_id` for other module can hooks this

cc @kittiu

### CODE REVIEW

File: hr_expense_advance_clearing/models/hr_expense_sheet.py [hunk 1]
```diff
@@ -138,15 +138,22 @@ def open_clear_advance(self):
...
     def _onchange_advance_sheet_id(self):
         self.expense_line_ids -= self.expense_line_ids.filtered("av_line_id")
         self.advance_sheet_id.expense_line_ids.sudo().read()  # prefetch
-        for line in self.advance_sheet_id.expense_line_ids.filtered(
-            "clearing_product_id"
-        ):
-            clear_advance = self._prepare_clear_advance(line)
-            self.expense_line_ids += self.env["hr.expense"].new(clear_advance)
+        lines = self.get_domain_advance_sheet_expense_line()
+        self.expense_line_ids = [
```
Comment (kittiu) [inline hr_expense_advance_clearing/models/hr_expense_sheet.py]: Not `+=` ?

Comment (kittiu) [reply hr_expense_advance_clearing/models/hr_expense_sheet.py]: The original code, what it try to do is to add lines, not assign lines, isn't it?
(says, user can also manually add some line first, before trigger onchange)

Comment (Saran440) [reply hr_expense_advance_clearing/models/hr_expense_sheet.py]: Okay, I forget this case.
Fixed.

Thank you for your reviewed.

**Review by kittiu — APPROVED** [review]

**Pull request merged**